ui/src/app/shared/components/webcomponents/confirmation-modal.component.ts

```diff
@@ -72,4 +121,33 @@ export class ConfirmationModalComponent {
     this.confirmText.valueChanges.subscribe(() => this.isValid = this.confirmText.value.toLocaleLowerCase() == 'delete');
 
   }
+  ngOnInit(){
+  }
+
+
+
+  openLinkedEntity(linkedEntity) {
+    let entityUrl;
+    if(this.modalData?.entityUrl)
+      entityUrl = this.formatEntityUrlIfExists(this.modalData?.entityUrl, linkedEntity);
+    if(!entityUrl) {
+      entityUrl = "/ui/td/cases/" + linkedEntity?.id + "/steps";
+    }
+    window.open(window.location.origin + entityUrl+"?stepGroupId="+this.modalData?.testCaseId, "_blank");
+  }
+
+  formatEntityUrlIfExists(url,linkedEntity){
+    return this.replaceParamValues(url, {entityId: linkedEntity?.id, versionId: linkedEntity?.workspaceVersionId})
+  }
+
+  replaceParamValues(url:string,data:object){
```

Comment:
plz move this to utitilities

Reply:
the function replaceParamValues is used only in [confirmation-modal.component.ts]. even in cloud it is present in the file and not under utilities

Reply:
it's okay if, it's not under utitlities in cloud. this is very generic function which can be reusable in nature